### PR TITLE
Feature/arc 454 meervoudig object slider

### PR DESCRIPTION
**Pull request by BavoVdg**
[ARC-454](https://meemoo.atlassian.net/browse/ARC-454)

⚠️ All current comments are fixed. 
- tab gedrag: https://studiohyperdrive.slack.com/archives/C0279QSL9CZ/p1647858086663889?thread_ts=1647851334.813449&cid=C0279QSL9CZ

# Meervoudig object slider
* Slider component
* Slider wordt enkel getoond als er meerdere representations zijn

**Testing**
- Uncomment `lijn 93 - 98` in `src/pages/leeszaal/[readingRoomSlug]/[objectId]/index.tsx` om enkele representations te mocken.
- Latest proxy env vars (`TICKET_SERVICE_URL`)

image

### CODE REVIEW

**Comment by bertyhell** [comment]
Ik test met deze url:
http://localhost:3200/leeszaal/leeszaal-8/x921c4s60t

maar ik zie zelfs na uncommenten van de regels geen bottom bar.
image

De flowplayer lijkt ook niet goed te initialiseren.

**Comment by bertyhell** [comment]
pixel offset op het linker vs het rechter paneel. Het linker panneel komt 1 pixel hoger dan het rechter.
image

File: src/modules/media/components/FragmentSlider/FragmentSlider.module.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		grid-template-columns: min-content 1fr min-content;
+		align-items: center;
+		overflow: hidden;
+	}
+
+	&__items {
+		display: grid;
+		grid-row-start: 1;
+		grid-column: 1 / 4;
+	}
```
Comment (bertyhell) [inline src/modules/media/components/FragmentSlider/FragmentSlider.module.scss:23]:
```suggestion
	&__items {
		display: grid;
		grid-row-start: 1;
		grid-column: 1 / 4;
		padding: 0.6rem;
	}
```

image

**Comment by bertyhell** [comment]
When running through the component using the keyboard, the items are nicely highlighted :D
But when i get to the last item:
image

And i press tab again:
image

current behavior: The scrollable scrolls back to the left and the "slide right" button becomes active and focused

expected behavior: the "slide right" button is not focusable, since it is greyed out and the focus should just skip the button. And the slider should stay scrolled to the right.

**Review by bertyhell — CHANGES_REQUESTED** [review]

**Comment by BavoVdg** [comment]
> When running through the component using the keyboard, the items are nicely highlighted :D But when i get to the last item: image
> 
> And i press tab again: image
> 
> current behavior: The scrollable scrolls back to the left and the "slide right" button becomes active and focused
> 
> expected behavior: the "slide right" button is not focusable, since it is greyed out and the focus should just skip the button. And the slider should stay scrolled to the right.

Expected behaviour misschien eens bespreken op een daily.

**Comment by bertyhell** [comment]
> Can not reproduce.

My env ticket service url in my proxy was not set correctly. Now it works.

**Comment by bertyhell** [comment]
Height of the video player should be limited to the available height - height of the fragments bar (if the fragments bar is visible)
image


 
image
image

File: src/modules/media/const/index.tsx [hunk 2]
```diff
@@ -8,6 +8,13 @@ import { Icon } from '@shared/components';
 
 import { DynamicActionMenuProps } from '../components/DynamicActionMenu';
 
+/**
+ * Render media
+ */
+
+export const flowplayerFormats: string[] = ['mp4', 'mp3'];
+export const imageFormats: string[] = ['png'];
```
Comment (bertyhell) [inline src/modules/media/const/index.tsx:16]: https://flowplayer.com/resources/guides/players/supportedformats
```suggestion
export const flowplayerFormats: string[] = ['mp4', 'mp3', 'm4a', 'ogv', 'ogg', 'webm', 'm3u8'];
export const imageFormats: string[] = ['png', 'jpg', 'jpeg', 'gif'];
```

**Review by bertyhell — APPROVED** [review]

File: next.config.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ module.exports = withTM({
 		tsconfigPath: './tsconfig.build.json',
 	},
 	images: {
-		domains: ['assets.viaa.be'],
+		domains: ['assets.viaa.be', 'via.placeholder.com'],
```
Comment (andrych) [inline next.config.js:38]: Is die `via.placeholder.com` nog nodig of was deze enkel voor te testen?

Comment (BavoVdg) [reply next.config.js:38]: In principe enkel om te testen. Blijft er idealiter in tot dat we thumbnails kunnen renderen. ik zet er een TODO bij.

File: src/modules/media/components/FragmentSlider/FragmentSlider.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+				setOffset(0);
+			} else if (updatedNeedsScrolling && offset !== active) {
+				setOffset(active);
+			}
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [fragmentsSize, totalFragments]);
+
+	return (
+		<>
```
[the remark this answers is not in the document]
Comment (BavoVdg) [reply src/modules/media/components/FragmentSlider/FragmentSlider.tsx:59]: Niet meer.

**Review by andrych — APPROVED** [review]

**Review by ian-emsens-shd — APPROVED** [review]

**Pull request merged**